src/shipit_code_coverage_backend/shipit_code_coverage_backend/v2/diff.py

```diff
@@ -0,0 +1,200 @@
+    }
+
+
+async def changes_on_files(files, push, revision_index, branch_name):
+    '''
+    List all the operations that happened on some files, in a push
+    '''
+    assert isinstance(files, (list, tuple))
+    assert isinstance(push, int)
+    query = {
+        'query': {
+            'bool': {
+
+                # Files intersection
+                'should': [
+                    {'match': {'changeset.files': f}}
+                    for f in files
+                ],
+
+                'must': [
+                    # After our commit
+                    {'range': {
+                        'index': {'gt': revision_index},
+                    }},
+
+                    # Same push and repo
+                    {'match': {'push.id': push}},
+                    {'match': {'branch.name': branch_name}},
+                ]
+            }
+        },
+
+        # Only load moves
+        '_source': ['changeset.moves', 'changeset.description'],
+
+        # Probably should use scan/scroll
+        'size': 100,
+    }
+    try:
+        others = await active_data_async.search('commits_above', body=query, index='repo')
+    except NoResults:
+        return {}
+
+    # Group by files, with only our interesting files
+    return {
+        filename: [
+            move['changes']
+            for other in others['hits']['hits']
+            for move in other['_source']['changeset']['moves']
+            if move['new']['name'].endswith(filename) or move['old']['name'].endswith(filename)
+        ]
+        for filename in files
+    }
+
+
+def rewind(all_changes, original_lines):
+
+    # TODO: move offsets calc in changes_on_files ?
+    offsets = {}
+    for changes in all_changes:
+        for change in changes:
+            if change['line'] not in offsets:
+                offsets[change['line']] = 0
+
+            offsets[change['line']] += change['action'] == '+' and 1 or -1
+
+    # Apply the sum of all changes above the line to each line
+    return {
+        line: line + sum([
+            offsets.get(i, 0)
+            for i in range(1, line)
+        ])
+        for line in original_lines
+    }
+
+
+async def coverage_diff(changeset):
+    '''
+    List all the coverage changes introduced by a diff
+    '''
+    # Load revision from MC
+    rev = active_data.get_changeset(changeset)
+    files = rev['changeset']['files']
+    push = rev['push']['id']
+    logger.info('Looking up coverage', files=files, push=push)
+
+    # Get the lines affected by the patch, per files
+    lines = {
+        filename: set(itertools.chain(*[
+            [c['line'] for c in move['changes']]
+            for move in rev['changeset']['moves']
+            if move['new']['name'].endswith(filename) or move['old']['name'].endswith(filename)
+        ]))
+        for filename in files
+    }
+
+    changes, coverage = await asyncio.gather(
+        # Load all the other commits on this push, above this changeset
```

Comment:
Same here, the push might not be the same as the patch, it could be a following push.